porter/v1/cluster_control_plane.proto

```diff
@@ -67,7 +67,17 @@ message PreflightCheckRequest {
 
 message PreflightCheckResponse {
   // preflight_checks keys are name of preflight check and error will be if the preflight checks fails, nil if it passes
-  map<string, Error> preflight_checks = 1;
+  // Deprecated: use preflight_check_errors instead.
+  map<string, Error> preflight_checks = 1 [deprecated = true];
+  // preflight_check_errors is a list of failing preflight checks
+  repeated PreflightCheckError preflight_check_errors = 2;
+}
+
+message PreflightCheckError {
+  // key is the key of the preflight check
+  string key = 1;
```

Comment:
do we want this to be an enum?

Reply:
yes

Reply:
An enum will prohibit any cloud-specific checks, without creating a mess. Would argue that its overoptimization